### ISSUE TITLE
Programs in the MIDI generated by `defaultMain` do not look right.

### ISSUE THREAD

**Issue by kindaro** [music-suite/music-suite#91]
## Description of the problem.

I hacked together some code that plays a few notes in each of a range of MIDI programs. Unfortunately, what I hear does not sound like the right MIDI programs get activated. Visual inspection of MIDI file confirms this suspicion.

## See for yourself.

This is my code:

```
music ∷ Music
music = foldr (</>) mempty . fmap (stretch (1/16)) $ zipWith delay [1 ..] $ concat $ flip fmap [20.. 30] \ channel → flip fmap [c, d, e, f, g] \ note → set parts' (solo (fromMidiProgram channel)) note
```

The command line: `cabal run sportif -- --filetype mid --out sportif-3.midi`

The MIDI file is attached: [`sportif-3.midi.tar.gz`](https://github.com/music-suite/music-suite/files/7775064/sportif-3.midi.tar.gz)

<details>
<summary>This is the view of the MIDI file as parsed by the package <code>midi</code>.</summary>

```
λ import Text.Show.Pretty 
λ fromFile "sportif-3.midi" >>= pPrint
Cons
  Parallel
  (Ticks 1024)
  [ 0 /. MetaEvent (SetTempo 1000000) ./ empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 22 })
          } ./
      1408 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 0 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 23 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 23 })
          } ./
      256 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 23 })
          } ./
      768 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 23 })
          } ./
      2048 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 1 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 26 })
          } ./
      2688 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 26 })
          } ./
      3328 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  , 0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody = Voice (ProgramChange Program { fromProgram = 26 })
          } ./
      3968 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 60 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 62 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 64 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 65 } Velocity { fromVelocity = 64 })
          } ./
      0 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOn Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      128 /.
      MIDIEvent
        Cons
          { messageChannel = Channel { fromChannel = 2 }
          , messageBody =
              Voice
                (NoteOff Pitch { fromPitch = 67 } Velocity { fromVelocity = 64 })
          } ./
      empty
  ]
```

</details>

## What I expected.

I expected to hear and see all MIDI programs from 20 to 30 iterated as specified.

## Discussion.

I find it hard to determine the cause of this problem. The code of `music-suite` is not trivial. Furthermore, I have little knowledge of MIDI — maybe I am missing something, maybe my MIDI emulator is not set up correctly, maybe my sound font is at fault.

**Comment by hanshoglund** [comment]
Note that the `</>` operator modifies parts. Since explicitly assign parts with `set parts` simple composition with `<>` should work.

**Comment by hanshoglund** [comment]
I've added a regression test similar to this in #92 which plays normally for me. Will look into reproducing it exactly.

**Comment by hanshoglund** [comment]
I've pushed a fix in #96. Note that we currently don't render playing techniques like pizzicato and guitar harmonics/dist in Midi output (see #98 #99).

@kindaro if you'd like to review, please ping me and I'll invite you.
